app/services/discovery_report.rb

```diff
@@ -19,6 +19,35 @@ def initialize(batch)
     @summary = { objects_with_error: 0, mimetypes: Hash.new(0), start_time: Time.now.utc.to_s, total_size: 0 }
   end
 
+  # @return [String] output_path to store report results, generate a different string each time
+  def output_path
+    Dir::Tmpname.create(["#{self.class.name.underscore}_", '.json'], batch.batch_context.output_dir) { |path| path }
+  end
+
+  # @return [Boolean]
+  def using_file_manifest?
+    using_file_manifest && File.exist?(File.join(bundle_dir, batch.batch_context.file_manifest))
+  end
+
+  # @return [PreAssembly::FileManifest]
+  def file_manifest
+    @file_manifest ||= PreAssembly::FileManifest.new(csv_filename: batch.batch_context.file_manifest, bundle_dir: bundle_dir)
+  end
+
+  # By using jbuilder on an enumerator, we reduce memory footprint (vs. to_a)
+  # @return [Jbuilder] (caller needs obj.to_builder.target! for the JSON string)
+  def to_builder
+    json_report = Jbuilder.new do |json|
+      json.rows { json.array!(each_row) }
+      json.summary summary.merge(end_time: Time.now.utc.to_s)
+    end
+    @objects_had_errors = (summary[:objects_with_error] > 0) # indicate if any objects generated errors
+    @error_message = "#{summary[:objects_with_error]} objects had errors in the discovery report" if objects_had_errors
+    json_report
+  end
```

Comment:
I don't think these need to become public again, do they?

Reply:
i think all are just staying public.  i did a cursory `grep`, and all of these methods appear to be called from outside this class.  the diff is a little wonky, but the change is making all the other methods private, as i don't think anything in this class was a private method before this PR.

Reply:
i think of this diff as moving all the other methods below the (new) `private` scope modifier, but the diff presents it as the public methods being moved above `private`.

Reply:
I don't believe these methods need to be public as none of the changes here seem to call them directly.  Is it possible you did this PR without rebasing on master?  The tests can use `.send(:method)`.

Reply:
i don't think it's a rebase issue -- the current `main` version of `discovery_report.rb` doesn't seem to use the `private` keyword at all:  https://github.com/sul-dlss/pre-assembly/blob/a64d56debaeb038a9b4368ecdc26693e4e39436d/app/services/discovery_report.rb (i linked to what's currently the most recent commit so that this comment stays linked to the same thing, but that commit is what the `HEAD` of `main` is at, afaict)

i'm also pretty sure i rebased this work on friday, and when i rebase on the latest `main` again now, then diff my local vs what's in this PR, the only difference i see is today's `rubocop-rails` dependency update.

looking again at the grep more closely, i'm not sure either `using_file_manifest?` or `file_manifest` are actually called anywhere.  but `output_path` and `to_builder` are both called in `discovery_report_job.rb`.

maybe i'm turned around on the rebase front.  but my current read is that while there's probably more room for cleanup, this makes some methods private where none in the class were private before, and so it seems like an improvement on the current code.

Reply:
no worries!  sorry if i left an unnecessary reply, i think github UI wasn't showing me the latest comments when i last commented.